lenses/tests/test_nslcd.aug

```diff
@@ -0,0 +1,408 @@
+  { "#comment" = "for details." }
+  {  }
+  { "#comment" = "Specifies the number of threads to start that can handle requests and perform LDAP queries." }
+  { "threads" = "5" }
+  {  }
+  { "#comment" = "The user and group nslcd should run as." }
+  { "uid" = "nslcd" }
+  { "gid" = "nslcd" }
+  {  }
+  { "#comment" = "This option controls the way logging is done." }
+  { "log" = "syslog info" }
+  {  }
+  { "#comment" = "The location at which the LDAP server(s) should be reachable." }
+  { "uri" = "ldaps://XXX.XXX.XXX" }
+  {  }
+  { "#comment" = "The search base that will be used for all queries." }
+  { "base" = "dc=XXX,dc=XXX" }
+  {  }
+  { "#comment" = "The LDAP protocol version to use." }
+  { "ldap_version" = "3" }
+  {  }
+  { "#comment" = "The DN to bind with for normal lookups." }
+  { "binddn" = "cn=annonymous,dc=example,dc=net" }
+  { "bindpw" = "secret" }
+  {  }
+  {  }
+  { "#comment" = "The DN used for password modifications by root." }
+  { "rootpwmoddn" = "cn=admin,dc=example,dc=com" }
+  {  }
+  { "#comment" = "The password used for password modifications by root." }
+  { "rootpwmodpw" = "XXXXXX" }
+  {  }
+  {  }
+  { "#comment" = "SASL authentication options" }
+  { "sasl_mech" = "OTP" }
+  { "sasl_realm" = "realm" }
+  { "sasl_authcid" = "authcid" }
+  { "sasl_authzid" = "dn:cn=annonymous,dc=example,dc=net" }
+  { "sasl_secprops" = "noanonymous,noplain,minssf=0,maxssf=2,maxbufsize=65535" }
+  { "sasl_canonicalize" = "yes" }
+  {  }
+  { "#comment" = "Kerberos authentication options" }
+  { "krb5_ccname" = "ccname" }
+  {  }
+  { "#comment" = "Search/mapping options" }
+  {  }
+  { "#comment" = "Specifies the base distinguished name (DN) to use as search base." }
+  { "base" = "dc=people,dc=example,dc=com" }
+  { "base" = "dc=morepeople,dc=example,dc=com" }
+  { "base"
+    { "alias" = "dc=aliases,dc=example,dc=com" }
+  }
+  { "base"
+    { "alias" = "dc=morealiases,dc=example,dc=com" }
+  }
+  { "base"
+    { "group" = "dc=group,dc=example,dc=com" }
+  }
+  { "base"
+    { "group" = "dc=moregroup,dc=example,dc=com" }
+  }
+  { "base"
+    { "passwd" = "dc=users,dc=example,dc=com" }
+  }
+  {  }
+  { "#comment" = "Specifies the search scope (subtree, onelevel, base or children)." }
+  { "scope" = "sub" }
+  { "scope"
+    { "passwd" = "sub" }
+  }
+  { "scope"
+    { "aliases" = "sub" }
+  }
+  {  }
+  { "#comment" = "Specifies the policy for dereferencing aliases." }
+  { "deref" = "never" }
+  {  }
+  { "#comment" = "Specifies whether automatic referral chasing should be enabled." }
+  { "referrals" = "yes" }
+  {  }
+  { "#comment" = "The FILTER is an LDAP search filter to use for a specific map." }
+  { "filter"
+    { "group" = "(objectClass=posixGroup)" }
+  }
+  {  }
+  { "#comment" = "This option allows for custom attributes to be looked up instead of the default RFC 2307 attributes." }
+  { "map"
+    { "passwd"
+      { "homeDirectory" = "\"${homeDirectory:-/home/$uid}\"" }
```

Comment:
I guess we can't remove the quotes for this specific field, as they are part of the value itself:

(from the documentation @ https://arthurdejong.org/nss-pam-ldapd/nslcd.conf.5)

> map MAP ATTRIBUTE NEWATTRIBUTE
> This option allows for custom attributes to be looked up instead of the default RFC 2307 attributes. The MAP may be one of the supported maps below. The ATTRIBUTE is the one as used in RFC 2307 (e.g. userPassword, ipProtocolNumber, macAddress, etc.). The NEWATTRIBUTE may be any attribute as it is available in the directory.
> If the NEWATTRIBUTE is presented in quotes (") it is treated as an expression which will be evaluated to build up the actual value used. See the section on attribute mapping expressions below for more details.